integration-tests/awsguard.go

```diff
@@ -0,0 +1,137 @@
+// policy pack to be used for the integration test. This is how any specific configuration
+// values can be used for the test. (e.g. disabling or configuring rules.)
+type awsGuardSettings struct {
+	// Enforcement level to use for all policies. ("" will default to "mandatory".)
+	defaultEnforcementLevel string
+
+	// Specific rules to disable. (Will set their individual enforcement levels to "disabled".)
+	disableRules []string
+}
+
+// validate confirms the settings present are reasonable. Since we are writing these settings
+// into JS code that gets executed, a malicious user could do some pretty nasty things without
+// safeguards in place.
+func (settings awsGuardSettings) validate() error {
+	// default enforcement level
+	switch settings.defaultEnforcementLevel {
+	case "", "advisory", "mandatory", "disabled":
+		// OK
+		break
+	default:
+		return errors.Errorf("unrecognized default enforcement level %q", settings.defaultEnforcementLevel)
+	}
+
+	// disabled rules
+	for _, rule := range settings.disableRules {
+		if !ruleNameRE.MatchString(rule) {
+			return errors.Errorf("rule name %q appears to be invalid", rule)
+		}
+	}
+
+	return nil
+}
+
+// CreatePolicyPack creates a new Node module in a sub folder of the test environment.
+// The awsGuardSettings will be written into the module's index.ts file.
+// Returns the path to the created module's directory.
+func (settings awsGuardSettings) CreatePolicyPack(e *ptesting.Environment) (string, error) {
+	e.Log("Creating customized AWS Guard module")
+
+	if err := settings.validate(); err != nil {
+		return "", errors.Wrap(err, "validation error")
+	}
+
+	initialCWD := e.CWD
+
+	moduleFolder := path.Join(e.RootPath, "custom-awsguard")
+	if err := os.Mkdir(moduleFolder, os.ModeDir|os.ModePerm); err != nil {
+		return "", errors.Wrap(err, "creating folder for customized AWS Guard module")
+	}
+
+	// package.json, defining the module itself.
+	packageJSONFilePath := path.Join(moduleFolder, "package.json")
+	packageJSONFileContents := `{
+		"name": "custom-awsguard",
+		"version": "1.0.0",
+		"description": "Customized AWS Guard policy pack for integration tests.",
+		"main": "index.js",
+		"dependencies": {
+		  "@pulumi/awsguard": "latest"
+		}
+	  }`
+	if err := ioutil.WriteFile(packageJSONFilePath, []byte(packageJSONFileContents), os.ModePerm); err != nil {
+		return "", errors.Wrap(err, "writing package.json")
+	}
+
+	// index.js, which contains encodes the settings via code.
+	indexTsFilePath := path.Join(moduleFolder, "index.js")
+	indexTsFileContents := settings.renderIndexJSFile()
+	if err := ioutil.WriteFile(indexTsFilePath, []byte(indexTsFileContents), os.ModePerm); err != nil {
+		return "", errors.Wrap(err, "writing index.js")
+	}
+
+	// Run `yarn` in the custom policy packs folder, to download the AWS Guard module as
+	// a dependency. (Which should have been "linked" with the development version in this
+	// repository.)
+	e.CWD = moduleFolder
+	e.RunCommand("yarn")
```

Comment:
I don't think it will automagically use the "linked" version. You have to explicitly run `yarn link @pulumi/awsguard` to use the linked version. I would run two explicit commands:

```go
e.RunCommand("yarn", "install")
e.RunCommand("yarn", "link", "@pulumi/awsguard")
```

And do we need to do `abortIfFailed(t)` after each `RunCommand` like we do in `integration_test.go`?

Reply:
Good catch! Done.